src/lib/comms/sol-bluetooth-impl-bluez.c

```diff
@@ -0,0 +1,981 @@
+    int r;
+
+    d = find_device_by_path(ctx, path);
+    SOL_NULL_CHECK_GOTO(d, skip);
+
+    info = &d->info;
+
+    r = sd_bus_message_read_basic(m, 's', &name);
+    SOL_INT_CHECK_GOTO(r, < 0, skip);
+
+    free(info->name);
+
+    /* When we receive the property again we will try to allocate again */
+    info->name = strdup(name);
+
+    return true;
+
+skip:
+    r = sd_bus_message_skip(m, "s");
+    SOL_INT_CHECK(r, < 0, false);
+
+    return false;
+}
+
+static bool
+device_property_paired_set(void *data, const char *path, sd_bus_message *m)
+{
+    struct context *ctx = data;
+    struct device_info *d;
+    struct sol_bt_device_info *info;
+    bool paired;
+    int r;
+
+    d = find_device_by_path(ctx, path);
+    SOL_NULL_CHECK(d, false);
+
+    info = &d->info;
+
+    r = sd_bus_message_read_basic(m, 'b', &paired);
+    SOL_INT_CHECK(r, < 0, false);
+
+    info->paired = paired;
+
+    return true;
+}
+
+static bool
+device_property_connected_set(void *data, const char *path, sd_bus_message *m)
+{
+    struct context *ctx = data;
+    struct device_info *d;
+    struct sol_bt_device_info *info;
+    bool connected;
+    int r;
+
+    d = find_device_by_path(ctx, path);
+    SOL_NULL_CHECK(d, false);
+
+    info = &d->info;
+
+    r = sd_bus_message_read_basic(m, 'b', &connected);
+    SOL_INT_CHECK(r, < 0, false);
+
+    info->connected = connected;
+
+    return true;
+}
+
+static bool
+device_property_uuids_set(void *data, const char *path, sd_bus_message *m)
+{
+    struct context *ctx = data;
+    struct device_info *d;
+    struct sol_bt_device_info *info;
+    struct sol_bt_uuid *u = NULL;
+    const char *uuid = NULL;
+    int r;
+
+    d = find_device_by_path(ctx, path);
+    SOL_NULL_CHECK_GOTO(d, skip);
+
+    info = &d->info;
+
+    sol_vector_clear(&info->uuids);
+
+    r = sd_bus_message_enter_container(m, 'a', "s");
+    SOL_INT_CHECK_GOTO(r, < 0, done);
+
+    while (sd_bus_message_read_basic(m, 's', &uuid) > 0) {
+        struct sol_str_slice slice;
+
+        u = sol_vector_append(&info->uuids);
+        SOL_NULL_CHECK_GOTO(u, done);
+
+        /* In practice, all UUIDs will be 36 bytes long, just being careful. */
+        slice = sol_str_slice_from_str(uuid);
+
+        r = sol_bt_uuid_from_str(u, &slice);
+        SOL_INT_CHECK_GOTO(r, < 0, done);
+    }
+
+    r = sd_bus_message_exit_container(m);
+
+done:
+    if (!u || r < 0)
+        sol_vector_clear(&info->uuids);
+
+    return true;
+
+skip:
+    r = sd_bus_message_skip(m, "as");
+    SOL_INT_CHECK(r, < 0, false);
+
+    return false;
+}
+
+static bool
+device_property_rssi_set(void *data, const char *path, sd_bus_message *m)
+{
+    struct context *ctx = data;
+    struct device_info *d;
+    struct sol_bt_device_info *info;
+    int16_t rssi;
+    int r;
+
+    d = find_device_by_path(ctx, path);
+    SOL_NULL_CHECK(d, false);
+
+    info = &d->info;
+
+    r = sd_bus_message_read_basic(m, 'n', &rssi);
+    SOL_INT_CHECK(r, < 0, false);
```

Comment:
Yeah, things are pretty inconsistent right now, I need to think a little more about this, and see how should we handle this kind of failure. Thanks.